apps/reference/_supabase_js/upgrade-guide.mdx

```diff
@@ -0,0 +1,406 @@
+---
+id: upgrade-guide
+title: Upgrade guide
+sidebar_label: Upgrade guide
+---
+
+import Tabs from '@theme/Tabs'
+import TabItem from '@theme/TabItem'
+
+# Upgrade guide
+
+Supabase.js upgrade guide.
+
+## Upgrading from version 1 to 2
+
+Supabase-js v2 is a major upgrade to the library with new api methods and a number of breaking changes.
+
+1. Update to the latest version:
+
+```sh
+npm install @supabase/supabase-js@2
+```
+
+2. *Optionally* if you are using custom configuration with `createClient` then follow below:
+
+<Tabs
+  groupId="version"
+  values={[
+    {label: 'Before', value: '1.x'},
+    {label: 'After', value: '2.x'},
+  ]}>
+
+<TabItem value="1.x">
+
+```ts title="src/supabaseClient.ts"
+const supabaseClient = createClient(SUPABASE_URL, SUPABASE_ANON_KEY, {
+  schema: 'custom',
+  persistSession: false
+})
+```
+</TabItem>
+
+<TabItem value="2.x">
+
+```ts title="src/supabaseClient.ts"
+const supabaseClient = createClient(SUPABASE_URL, SUPABASE_ANON_KEY, {
+  db: {
+    schema: 'custom'
+  },
+  auth: {
+    persistSession: true
+  }
+})
+```
+</TabItem>
+</Tabs>
+
+You can read more about the constructor options in the [release notes](http://localhost:3010/docs/reference/javascript/release-notes#explicit-constructor-options).
+
+### Auth methods
+
+#### Signing in with email and password
```

Comment:
I think the header change didn't look well when the site was rendered so I reverted them. I wasn't too sure about the adding of context since the release notes kinda already have some, but I will look into adding some in the morning when I get up.